pallets/asset/src/lib.rs

```diff
@@ -740,30 +742,37 @@ decl_module! {
             let mut issued_in_this_round = Self::issued_in_funding_round(&ticker_round);
 
             // A round of per-investor checks
-            for i in 0..issue_asset_items.len() {
+            for IssueAssetItem { investor_did, value } in &issue_asset_items {
                 ensure!(
-                    Self::check_granularity(&ticker, issue_asset_items[i].value),
+                    Self::check_granularity(&ticker, *value),
                     Error::<T>::InvalidGranularity
                 );
                 let updated_total_supply = token
                     .total_supply
-                    .checked_add(&issue_asset_items[i].value)
+                    .checked_add(value)
                     .ok_or(Error::<T>::TotalSupplyOverflow)?;
                 ensure!(updated_total_supply <= MAX_SUPPLY.into(), Error::<T>::TotalSupplyAboveLimit);
 
-                current_balances.push(Self::balance(&ticker, &issue_asset_items[i].investor_did));
-                updated_balances.push(current_balances[i]
-                    .checked_add(&issue_asset_items[i].value)
-                    .ok_or(Error::<T>::BalanceOverflow)?);
+                let bals = Self::balance(&ticker, *investor_did);
+                current_balances.push(bals);
+                let updated_total_balance = bals
+                    .0
+                    .checked_add(value)
+                    .ok_or(Error::<T>::BalanceOverflow)?;
+                let updated_def_balance = bals
+                    .1
+                    .checked_add(value)
+                    .ok_or(Error::<T>::DefaultPortfolioBalanceOverflow)?;
```

Comment:
This check is redundant since individual portfolio balance will always be equal or less than the total portfolio balance.